File: src/neetcode/Java/_14LongestCommonPrefix.java
package neetcode.Java;

public class _14LongestCommonPrefix {
    public static String longestCommonPrefix(String[] strs) {
        int lenStrs = strs.length;
        int i = 0;
        int len = strs[0].length();
        Task:
        while (i < len) {
            for (int j = 1; j < lenStrs; j++) {
                if (i >= strs[j].length() || strs[0].charAt(i) != strs[j].charAt(i)) break Task;
            }
            i++;
        }
        return strs[0].substring(0, i);
    }

    public static void main(String[] args) {
        String[] s = {"dog","racecar","car"};
        String res = longestCommonPrefix(s);
        System.out.println(res);
    }
}

File: src/neetcode/kotlin/_14LongestCommonPrefix.kt
package neetcode.kotlin

object _14LongestCommonPrefix {
    fun longestCommonPrefix(strs: Array<String>): String {
        val lenStrs = strs.size
        var i = 0
        val len = strs[0].length
        Task@ while (i < len) {
            for (j in 1 until lenStrs) {
                if (i >= strs[j].length || strs[0][i] != strs[j][i]) break@Task
            }
            i++
        }
        return strs[0].substring(0, i)
    }

    @JvmStatic
    fun main(args: Array<String>) {
        val s = arrayOf("dog", "racecar", "car")
        val res = longestCommonPrefix(s)
        println(res)
    }
}
